common/document_parser/lib/responsibility_parse/responsibility_parser.py

```diff
@@ -0,0 +1,335 @@
+                            new_role_start_metadata["n_periods"] == numbering.count(".") and \
+                            new_role_start_metadata["n_parenthesis"] == numbering.count(")") and \
+                            new_role_start_metadata["n_numbers"] <= sum(c.isdigit() for c in numbering) and \
+                            new_role_start_metadata["n_letters"] <= sum(
+                        c.isalpha() for c in numbering):  # and \
+                        # resp_line.endswith(self.new_role_find_character):
+                        if section_resp_lines:
+                            responsibility_section_list.append(section_resp_lines)
+                            section_resp_lines = []
+                    section_resp_lines.append(resp_line)
+
+            ## determine whether the current line (w/o any numbering/punctuation) is a continuation of an existing line
+            # or a new line that should be captured as a new line
+            elif section_resp_lines != []:
+                ## if the previous line ended in punctuation, then add this text as a new line
+                if section_resp_lines[-1].strip().endswith(self.new_role_find_character):
+                    section_resp_lines.append(resp_line)
+                # no punctuation at end of previous line, so this line is a continuation of the previous line of text
+                else:
+                    section_resp_lines[-1] += f" {resp_line}"
+            ## determine whether there is a free text line/paragraph at the beginning of the responsibility section to add
+            else:
+                if " 1. " in resp_line:
+                    section_resp_lines.append(f"1.{resp_line.split('1.')[-1]}")
+                    new_role_start_metadata = {
+                        "n_periods": 1,
+                        "n_parenthesis": 0,
+                        "n_numbers": 1,
+                        "n_letters": 0,
+                    }
+
+                elif (entities_found_list and resp_line.strip().endswith(self.new_role_find_character)) or any(
+                        word in resp_line for word in self.new_role_key_words):
+                    section_resp_lines.append(resp_line)
+        if section_resp_lines:
+            responsibility_section_list.append(section_resp_lines)
+
+        return responsibility_section_list
+
+    def extract_responsibilities_from_json(self, json_filepath):
+        """
+        Takes a JSON dict and parses the responsibility section(s)
+        Args:
+            json_filepath: (dict) GC JSON filepath that will have responsibility sections parsed from it
+
+        Returns:
+
+        """
+        json_file_name = json_filepath.split("/")[-1]
+        try:
+            json_dict = json.load(open(json_filepath, "r"))
+            title = json_dict.get("title")
+            filename = json_dict.get("filename")
+            resp_sections = json_dict.get('sections',{}).get("responsibilities_section",[])
```

Comment:
In the future, you can use the [`FieldNames` class](https://github.com/dod-advana/gamechanger-data/blob/dev/common/document_parser/lib/document/field_names.py). It's helpful in case any field names change, so we don't have to update hard-coded strings :)

Reply:
Great idea, thanks for pointing this out - made the updates https://github.com/dod-advana/gamechanger-data/pull/252/commits/371ec092ff1df275f61986aaa05b8e6916a5fbb3